### PR TITLE
ROX-25446: Deprecate Central PVC from operator

### PR DESCRIPTION
**Pull request by c-du**
### Description

The second PR of Central PVC deprecation.
The CHANGELOG will be updated later.

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing

- [ ] inspected CI results

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

change me!

- [x] validated that central persistence does not show in operator UI:
image

### CODE REVIEW

File: operator/pkg/central/extensions/reconcile_pvc.go [hunk 1]
```diff
@@ -137,7 +131,8 @@ func (r *reconcilePVCExtensionRun) Execute() error {
 	if r.centralObj.DeletionTimestamp != nil || r.persistence == nil {
 		return r.handleDelete()
 	}
-	if common.ObsoletePVC(r.centralObj.GetAnnotations()) {
+	// Starting from 4.6, we do not create or own PVCTargetCentral
+	if r.target == PVCTargetCentral {
 		return r.handleDelete()
```
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_pvc.go:136]: This actually seems to be unreachable code, since for `target == PVCTargetCentral` the `getPersistenceByTarget` always returns `nil`, which causes the `handleDelete` on line 138 (132 in changed file) to run. Perhaps consider folding this into the `if` above? 🤔

File: operator/pkg/central/extensions/reconcile_pvc_test.go [hunk 2]
```diff
@@ -81,71 +81,40 @@ func notOwnedBy(central *platform.Central) pvcVerifyFunc {
 }
 
 func TestReconcilePVCExtension(t *testing.T) {
-	removedCentral := makeCentral(nil)
-	emptyNotDeletedCentral := makeCentral(nil)
-	emptyNotDeletedCentralWithDB := makeCentral(nil)
-	emptyNotDeletedCentralWithDB.Spec.Central.DB = &platform.CentralDBSpec{}
+	emptyCentral := makeCentral(nil)
+	emptyCentralWithDB := makeCentral(nil)
+	emptyCentralWithDB.Spec.Central.DB = &platform.CentralDBSpec{}
```
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_pvc_test.go:86]: What is the purpose of this line? It seems to replace a pointer to an empty struct with a pointer to another empty struct 🤔

Comment (porridge) [reply operator/pkg/central/extensions/reconcile_pvc_test.go:86]: And if so, and this line is not needed, then we don't need both emptyCentral and emptyCentralWithDB right?

**Review by porridge — CHANGES_REQUESTED** [review]
I love the red color 😍 
See comments inline.

File: operator/pkg/central/extensions/reconcile_pvc.go [hunk 3]
```diff
@@ -76,12 +75,7 @@ func convertDBPersistenceToPersistence(p *platform.DBPersistence) *platform.Pers
 func getPersistenceByTarget(central *platform.Central, target PVCTarget) *platform.Persistence {
 	switch target {
 	case PVCTargetCentral:
```
Comment (0x656b694d) [inline operator/pkg/central/extensions/reconcile_pvc.go:77]: Is this still a possible case?

File: operator/pkg/central/common/common.go [hunk 4]
```diff
@@ -1,13 +1,7 @@
 package common
 
-import "strings"
-
 const (
-	// CentralPVCObsoleteAnnotation represents Central PVC has been obsoleted
+	// CentralPVCObsoleteAnnotation represents Central PVC has been obsoleted.
+	// Only used for test
```
Comment (stehessel) [inline operator/pkg/central/common/common.go:5]:
```suggestion
	// CentralPVCObsoleteAnnotation represents that the Central PVC been made obsolete.
	// Only used in tests.
```

File: operator/pkg/central/values/translation/translation.go [hunk 5]
```diff
@@ -219,15 +173,6 @@ func getCentralComponentValues(c *platform.CentralComponentSpec, checker *pvcSta
 	cv.SetStringMap("nodeSelector", c.NodeSelector)
 	cv.AddAllFrom(translation.GetTolerations(translation.TolerationsKey, c.Tolerations))
 
-	// TODO(ROX-7147): design CentralEndpointSpec, see central_types.go
```
Comment (stehessel) [inline operator/pkg/central/values/translation/translation.go:222]: Why did you remove this TODO?

File: operator/pkg/central/extensions/reconcile_pvc.go (hunk 3 above)
Comment (c-du) [reply operator/pkg/central/extensions/reconcile_pvc.go:77]: Yes, we shall remove the owner for Central.

File: operator/pkg/central/values/translation/translation.go (hunk 5 above)
Comment (c-du) [reply operator/pkg/central/values/translation/translation.go:222]: ROX-7147 is closed. This comment do not seem to directly relate to the contexts. Do you want to keep it longer?

Comment (stehessel) [reply operator/pkg/central/values/translation/translation.go:222]: So seems fine to remove it, just wanted to make sure it wasn't accidental.

File: operator/pkg/central/extensions/reconcile_pvc.go [hunk 6]
```diff
@@ -137,7 +131,8 @@ func (r *reconcilePVCExtensionRun) Execute() error {
 	if r.centralObj.DeletionTimestamp != nil || r.persistence == nil {
 		return r.handleDelete()
 	}
-	if common.ObsoletePVC(r.centralObj.GetAnnotations()) {
```
[the remark this answers is not in the document]
Comment (porridge) [reply operator/pkg/central/extensions/reconcile_pvc.go]: Looks like it would have been disowned if the user placed the annotation there. I thought that was intentional 😅 
Probably no point making this an error at this point.

File: operator/pkg/central/extensions/reconcile_pvc_test.go (hunk 2 above)
Comment (porridge) [reply operator/pkg/central/extensions/reconcile_pvc_test.go]: I'm still not getting this 😅

File: operator/internal/central/extensions/reconcile_pvc_test.go [hunk 7]
```diff
@@ -81,71 +81,40 @@ func notOwnedBy(central *platform.Central) pvcVerifyFunc {
...
-			StorageClassName: pointer.String("new-storage-class"),
-			ClaimName:        pointer.String(testPVCName),
-		},
-	})
-	centralWithPersistenceAndPvcObsoletedAnnotation.Annotations = pvcObsoletedAnnotation
+	emptyCentralWithPvcObsoletedAnnotation := makeCentral(nil)
+	emptyCentralWithPvcObsoletedAnnotation.Annotations = pvcObsoletedAnnotation
 
 	externalCentralWithDB := makeCentral(nil)
 	externalCentralWithDB.Spec.Central.DB = &platform.CentralDBSpec{}
```
Comment (porridge) [inline operator/internal/central/extensions/reconcile_pvc_test.go:95]: I think this line is not needed now either, that `makeCentral` initializes the DB struct...

**Review by porridge — APPROVED** [review]
This mostly LGTM now, modulo a handful of nitpicks inline.

**Review by stehessel — APPROVED** [review]

File: operator/pkg/central/extensions/reconcile_pvc.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (c-du) [reply operator/pkg/central/extensions/reconcile_pvc.go]: Yes. And I made a fix for this for 4.5.2 and 4.4.5.  FYI: ROX-25836.
